### PR TITLE
Updating Supported Versions of macOS for Simulators

### PR DESCRIPTION
**Pull request by mp-sauce**
Preparing for deprecation of macOS 10.x across desktop and iOS Sims

### Description
Per [this announcement](https://changelog.saucelabs.com/en/end-of-life-notice-february-13-2025-macosx-operating-system-DnwGVaKQ), I updated areas where macOS 10.x was referenced for support for desktop, iOS Sims, and Safari.

Updated records from the following pages:
https://docs.saucelabs.com/mobile-apps/automated-testing/appium/appium-versions/#ios-simulators
https://docs.saucelabs.com/mobile-apps/automated-testing/appium/virtual-devices/#ios-code-examples
https://docs.saucelabs.com/mobile-apps/supported-devices/#system-requirements 
https://docs.saucelabs.com/secure-connections/sauce-connect/security-authentication/#when-to-disable-ssl-bumping
https://docs.saucelabs.com/basics/test-config-annotation/test-config/#macosxsafari
https://docs.saucelabs.com/web-apps/automated-testing/electron/#supported-testing-platforms

### Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation fix (typos, incorrect content, missing content, etc.)

### CODE REVIEW

**Review by mk-ahmed-saucelabs — APPROVED** [review]
LGTM

File: docs/web-apps/automated-testing/playwright.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ Cucumber.js is not directly supported by Playwright. However, Playwright can be
 Supported OS:
 
 - Windows 10 / Windows 11
-- macOS 10.15+
+- macOS 11+
```
Comment (alexplischke) [inline docs/web-apps/automated-testing/playwright.md:23]: This refers to the supported platform that saucectl runs on (client side!). _Not what's on Sauce Labs_.

1. Now, I personally don't care if you want to bump this too.
2. If this section is not clear enough with regards to what _System Requirements_ refers to, perhaps we should remove it altogether from this location 💭

Comment (mp-sauce) [reply docs/web-apps/automated-testing/playwright.md:23]: Ah, thanks for catching this. After your message yesterday I then realized I may have made the same mistake for the supported frameworks. I'll confirm on whether this section still makes sense for playwright and the other frameworks.

Comment (mp-sauce) [reply docs/web-apps/automated-testing/playwright.md:23]: Consensus was to stick to this change. thanks for highlighting!

**Review by alexplischke — APPROVED** [review]

**Pull request merged**